runtime/queries/c/highlights.scm

```diff
@@ -1,61 +1,98 @@
-(storage_class_specifier) @keyword.storage
-
-"goto" @keyword
-"register" @keyword
-"break" @keyword
-"case" @keyword
-"continue" @keyword
-"default" @keyword
-"do" @keyword
-"else" @keyword
-"enum" @keyword
-"extern" @keyword
-"for" @keyword
-"if" @keyword
-"inline" @keyword
-"return" @keyword
-"sizeof" @keyword
-"struct" @keyword
-"switch" @keyword
-"typedef" @keyword
-"union" @keyword
-"volatile" @keyword
-"while" @keyword
-"const" @keyword
+[
+  "sizeof"
+] @keyword
+
+[
+  "enum"
+  "struct"
+  "union"
+] @keyword.storage.type
+
+[
+  "const"
+  "extern"
+  "inline"
+  "register"
+  "typedef"
+  "volatile"
+  (storage_class_specifier)
+] @keyword.storage.modifier
+
+[
+  "for"
+  "do"
+  "while"
+  "break"
+  "continue"
+] @keyword.control.repeat
 
 [
- "#define"
- "#elif"
- "#else"
- "#endif"
- "#if"
- "#ifdef"
- "#ifndef"
- "#include"
- (preproc_directive)
+  "goto"
+  "if"
+  "else"
+  "switch"
+  "case"
+  "default"
+] @keyword.control.conditional
+
+"return" @keyword.control.return
+
+[
+  "defined"
+  "#define"
+  "#elif"
+  "#else"
+  "#endif"
+  "#if"
+  "#ifdef"
+  "#ifndef"
+  "#include"
+  (preproc_directive)
 ] @keyword.directive
 
-"--" @operator
-"-" @operator
-"-=" @operator
-"->" @operator
-"=" @operator
-"!=" @operator
-"*" @operator
-"&" @operator
-"&&" @operator
-"+" @operator
-"++" @operator
-"+=" @operator
-"<" @operator
-"==" @operator
-">" @operator
-"||" @operator
-">=" @operator
-"<=" @operator
-
-"." @punctuation.delimiter
-";" @punctuation.delimiter
+[
+  "+"
+  "-"
+  "*"
+  "/"
+  "++"
+  "--"
+  "%"
+  "=="
+  "!="
+  ">"
+  "<"
+  ">="
+  "<="
+  "&&"
+  "||"
+  "!"
+  "&"
+  "|"
+  "^"
+  "~"
+  "<<"
+  ">>"
+  "="
+  "+="
+  "-="
+  "*="
+  "/="
+  "%="
+  "<<="
+  ">>="
+  "&="
+  "^="
+  "|="
+  "->"
+  "::"
+  "?"
+  "..."
+] @operator
+
+["," "." ":" ";"] @punctuation.delimiter
+
+["(" ")" "[" "]" "{" "}"] @punctuation.bracket
 
 [(true) (false)] @constant.builtin.boolean
 
```

Comment:
On the next line there's a stanza

```
(enumerator) @type.enum.variant
```

which I think should be

```
(enumerator name: (identifier) @type)
```

to fix highlighting for enum variants like in [this block](https://github.com/mawww/kakoune/blob/24d6072353f7c7e7cac84b4eb085036a8c852f96/src/normal.cc#L32-L37)

Reply:
Also I enjoy the use of Kakoune's source code there 🤭